src/main/java/com/photolocos/enterprise/controller/PhotoLocosController.java

```diff
@@ -51,30 +51,47 @@ public ModelAndView savePhoto(@RequestParam("image") MultipartFile image, PhotoD
 
     @GetMapping("/photosByTag/{tag}/")
     public ModelAndView photosByTag(@PathVariable("tag") String tag) {
+        try {
         ModelAndView modelAndView = new ModelAndView();
         modelAndView.setViewName("index");
         Set<PhotoDTO> photos = photoService.fetchByTag(tag);
         modelAndView.addObject("photos", photos);
         return  modelAndView;
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+            return "error";
```

Comment:
caching an exception is a good idea, if:
1) There is a checked exception that could be thrown.  Not sure if that is the case here.
2) If the exception is properly logged.  e.printStackTrace is not proper logging, as it does not have the concept of log priority, and it only goes one place - the console, which no one will see.

So, a good start, but I'd add a bit more before considering this complete.